db/db_impl/db_impl_open.cc

```diff
@@ -445,8 +451,11 @@ Status DBImpl::Recover(
   if (!immutable_db_options_.best_efforts_recovery) {
     s = versions_->Recover(column_families, read_only, &db_id_);
   } else {
+    if (read_only) {  // TODO: get rid of this case?
+      files_status = env_->GetChildren(dbname_, &db_file_names);
+    }
     s = versions_->TryRecover(column_families, read_only, &db_id_,
-                              &missing_table_file);
+                              &missing_table_file, db_file_names, files_status);
```

Comment:
If `files_status` is non-ok, we do not call this. In fact, we should have already returned at line 422. Therefore, removing `files_status`.